src/QsCompiler/CompilationManager/FileContentManager.cs

```diff
@@ -32,6 +32,11 @@ public class FileContentManager : IDisposable
 
         public string FileName { get; }
 
+        /// <summary>
+        /// Indicates the current version of the document according to the Language Server Protocol.
+        /// </summary>
+        public int? Version { get; set; }
```

Comment:
I think it would be nice to enforce the invariant that the file version is monotonically increasing.

Reply:
Should we log an error or fail if that happens? Also, I'm not sure how to enforce it on our side, since we're just recording the value sent in 'didChange' notifications, and in the unlikely situation in which the IDE restarted the numbering, that would still be the version that we should use with the protocol. Assuming that it happens, I don't know if erroring out on our side would be better experience, than attempting to follow the protocol with the new value.

Would it be ok if we simply log this condition for debugging purposes?

Reply:
> in the unlikely situation in which the IDE restarted the numbering, that would still be the version that we should use with the protocol

According to the LSP specification, the client is not allowed to do this. If this happens, then I think it would indicate that the client and server are out of sync, and maybe reloading the file from disk (which would reset the version number) would be best.

Reply:
It is an interesting discussion to decide how permissive our server should be, given that the history of this bug started with the editors being very relaxed with the versioning protocol and then VS started enforcing it in 16.10. The risk of actively enforcing the version from our side is not minimal, and that would make the change out of the scope of a bug fix. However, we're discussing this outside of this PR, and we can decide what our future plans are.